controlplane/eks/controllers/awsmanagedcontrolplane_controller.go

```diff
@@ -285,8 +285,10 @@ func (r *AWSManagedControlPlaneReconciler) reconcileDelete(ctx context.Context,
 
 	if err := sgService.DeleteSecurityGroups(); err != nil {
 		log.Error(err, "error deleting general security groups for AWSManagedControlPlane", "namespace", controlPlane.Namespace, "name", controlPlane.Name)
+		conditions.MarkFalse(managedScope.InfraCluster(), infrav1.ClusterSecurityGroupsReadyCondition, clusterv1.DeletionFailedReason, clusterv1.ConditionSeverityWarning, err.Error())
```

Comment:
Deletion is not failed in all scenarios where DeleteSecurityGroups() returns error. 
Having the condition set as deleting and erroring is more accurate.
For example, `describeClusterOwnedSecurityGroups()` could fail and this doesn't mean deletion process failed.

Reply:
I agree with you that failed call to `describeClusterOwnedSecurityGroups` and failed call to `revokeAllSecurityGroupIngressRules` are different. 

However if we mark the condition as Deleting first, we could run into the same issue that this PR is trying to address (patching the same condition multiple times in one reconciliation loop). 

One option is to mark conditions explicitly inside `DeleteSecurityGroups` (e.g. set the reason to something like DescribeFailed if `describeClusterOwnedSecurityGroups` failed and set the reason to DeleteFailed if `revokeAllSecurityGroupIngressRules` failed) and does not mark the condition in `awsmanagedcontrolplane_controller.go`. WDYT?